engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/EqualsSimpleNode.java

```diff
@@ -0,0 +1,386 @@
+package org.enso.interpreter.node.expression.builtin.meta;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Cached.Shared;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.TruffleObject;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.Node;
+import java.math.BigInteger;
+import org.enso.interpreter.node.expression.builtin.number.utils.BigIntegerOps;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.EnsoMultiValue;
+import org.enso.interpreter.runtime.data.atom.Atom;
+import org.enso.interpreter.runtime.data.atom.AtomConstructor;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.error.WarningsLibrary;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+import org.enso.polyglot.common_utils.Core_Text_Utils;
+
+@GenerateUncached
+public abstract class EqualsSimpleNode extends Node {
+
+  public static EqualsSimpleNode build() {
+    return EqualsSimpleNodeGen.create();
+  }
+
+  public abstract boolean execute(VirtualFrame frame, Object self, Object right);
+
+  @Specialization
+  boolean equalsBoolBool(boolean self, boolean other) {
+    return self == other;
+  }
+
+  @Specialization
+  boolean equalsBoolDouble(boolean self, double other) {
+    return false;
+  }
+
+  @Specialization
+  boolean equalsBoolLong(boolean self, long other) {
+    return false;
+  }
+
+  @Specialization
+  boolean equalsBoolBigInt(boolean self, EnsoBigInteger other) {
+    return false;
+  }
+
+  @Specialization
+  boolean equalsBoolText(boolean self, Text other) {
+    return false;
+  }
+
+  @Specialization
+  boolean equalsBoolInterop(
+      boolean self,
+      Object other,
+      @Shared("interop") @CachedLibrary(limit = "10") InteropLibrary interop) {
+    try {
+      return self == interop.asBoolean(other);
+    } catch (UnsupportedMessageException ex) {
```

Comment:
Would it be more efficient to do the `isBoolean` check instead of recovering from the exception?